ios/RCTTWVideoModule.m

```diff
@@ -96,18 +98,302 @@ - (void)addParticipantView:(TVIVideoView *)view identity:(NSString *)identity  t
+                                [outgoingTrack setObject:roundTripTime forKey:@"roundTripTime"];
+                           }
+                           if([videoTrack codec]){
+                               NSString *codec = [NSString stringWithFormat:@"%@", [videoTrack codec]];
+                               [outgoingTrack setObject:codec forKey:@"codec"];
+                           }
+                           if([videoTrack frameRate]){
+                                  NSNumber *frameRate= [f  numberFromString:[NSString stringWithFormat:@"%lu", (unsigned long)[videoTrack frameRate]]];
+
+                               [outgoingTrack setObject:frameRate forKey:@"frameRate"];
+                           }
+                           if([videoTrack trackId]){
+                               trackId = [NSString stringWithFormat:@"%@", [videoTrack trackId]];
+                               [outgoingTrack setObject:trackId forKey:@"trackId"];
+                           }
+
+                           if([videoTrack packetsSent]){
+                               NSNumber *packetsSent= [f  numberFromString:[NSString stringWithFormat:@"%lu", (unsigned long)[videoTrack packetsSent]]];
+                               [outgoingTrack setObject:packetsSent forKey:@"packetsSent"];
+                           }
+
+                           if([videoTrack packetsLost]){
+                               NSNumber *packetsLost= [f  numberFromString:[NSString stringWithFormat:@"%lu", (unsigned long)[videoTrack packetsLost]]];
+                               [outgoingTrack setObject:packetsLost forKey:@"packetsLost"];
+                           }
+
+                           if([videoTrack valueForKey:@"dimensions"]){
+                               CMVideoDimensions dimensions =  [videoTrack dimensions];
+
+                               NSNumber *height= [f  numberFromString:[NSString stringWithFormat:@"%d", dimensions.height]];
+
+                               [outgoingTrack setObject:height forKey:@"height"];
+
+                               NSNumber *width= [f  numberFromString:[NSString stringWithFormat:@"%d", dimensions.width]];
+
+                               [outgoingTrack setObject:width forKey:@"width"];
+
+                           }
+
+                           if([videoTrack valueForKey:@"timestamp"]){
+
+                               NSNumber *timestamp= [f  numberFromString:[NSString stringWithFormat:@"%lu", (unsigned long)[videoTrack valueForKey:@"timestamp"]]];
+
+                               [outgoingTrack setObject:timestamp forKey:@"timestamp"];
+                           }
+
+                           [localVideoTracks setObject:outgoingTrack forKey:trackId];
+
+
+                       }
+                   }
+
+
+                   if([statReport audioTrackStats]){
+                       NSArray *audioTracks = [statReport audioTrackStats];
+                       for (id audioTrack in audioTracks) {
```

Comment:
Would be nice to pull some of this out to a helper function

Reply:
could definitely be condensed in a few ways. I kinda want to eventually get all the ice candidate data as well, but its a whole batch of stuff most people wont need or be able to do anything about.